scripts/pipelines/build_artifact

```diff
@@ -7,7 +7,7 @@ export PATH=${COMPOSER_BIN}:${PATH}
 # Generate artifact in a separate directory.
 export NEW_BUILD_DIR=/tmp/artifact
 blt deploy:check-dirty --ansi --verbose
-blt artifact:build -D deploy.dir=${NEW_BUILD_DIR} --ansi --verbose
+blt artifact:build -D deploy.dir=${NEW_BUILD_DIR} --environment=ci --ansi --verbose
```

Comment:
This is the second or third bug we've caught where this parameter was missing.

There are ways we could refactor this to avoid passing `--environment=ci` every time, such as setting `BLT_ENV=ci` as an environment variable once at the start of the script, or by incorporating `EnvironmentDetector::isCiEnv()` into the DefaultConfig environment detection step. They aren't without risk though, and at any rate are beyond the scope of this PR.

Reply:
CI=true is present in most CI environments today. If not (looking at Jenkins, people can set it).

Why does blt have to act differently in CI vs on your machine ?

Reply:
Good point about the CI variable, we should key off of that. I created this followup issue: https://github.com/acquia/blt/issues/3859

This ticket is itself a good example of why BLT needs to behave differently. Systems paths are difference, available resources are different, you need different modules enabled, etc... In this particular case, the deploy directory is pretty special in Pipelines.